From: Thomas Gleixner <tglx@linutronix.de>
Date: Sat, 25 Jul 2009 22:06:27 +0200
Subject: [PATCH 051/397] mm: Remove preempt count from pagefault
 disable/enable
Origin: https://git.kernel.org/cgit/linux/kernel/git/rt/linux-stable-rt.git/commit?id=84c4b3e3f6fd357b00212c806e99225d95184390

Now that all users are cleaned up, we can remove the preemption count.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 mm/memory.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/mm/memory.c b/mm/memory.c
index 856f7e357309..3ab5b52b5be6 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -3538,7 +3538,6 @@ unlock:
 #ifdef CONFIG_PREEMPT_RT_FULL
 void pagefault_disable(void)
 {
-	inc_preempt_count();
 	current->pagefault_disabled++;
 	/*
 	 * make sure to have issued the store before a pagefault
@@ -3556,12 +3555,6 @@ void pagefault_enable(void)
 	 */
 	barrier();
 	current->pagefault_disabled--;
-	dec_preempt_count();
-	/*
-	 * make sure we do..
-	 */
-	barrier();
-	preempt_check_resched();
 }
 EXPORT_SYMBOL_GPL(pagefault_enable);
 #endif
